lib/zendesk_api/pagination.rb

```diff
@@ -0,0 +1,107 @@
+module ZendeskAPI
+  # Contains all methods related to pagination in an attempt to slim down collection.rb
```

Comment:
Minor: If you wanted to be super-clear about the fact it's only for collection, you could even namespace it by `module Collection` here. That way, it wouldn't pollute the namespace in `ZendeskAPI` when it should only be used in `Collection`.